### PR TITLE
DNM Accelio submodule

### PR DESCRIPTION
**Pull request by mattbenjamin**
I had pushed this last week, should have created an immediate DNM PR.

### CODE REVIEW

**Comment by gregsfortytwo** [comment]
Not sure what you're looking for since it's currently DNM, but this has some merge conflicts now.

Also I think there was a discussion about why to add the submodule that'd be nice to summarize in the commit somewhere since I can't find it now. :)

**Comment by mattbenjamin** [comment]
I'll rebase it.

**Comment by mattbenjamin** [comment]
Sorry, in addition to finding the text from Sage (found it)

I wanted to get review--I can rename it if that's what is preferred?

**Comment by mattbenjamin** [comment]
Look ok?

**Comment by liewegas** [comment]
I don't really understand what "AC_SUBST(EXTRALIBS)" does but assuming this builds it looks good to me!

File: autogen.sh [hunk 1]
```diff
@@ -47,4 +47,5 @@ autoheader
 automake -a --add-missing -Wall
 ( cd src/gmock && autoreconf -fvi; )
 ( cd src/rocksdb && autoreconf -fvi; )
+( cd src/accelio && autoreconf -fvi; )
```
Comment (liewegas) [inline autogen.sh]: this either needs to be conditional on src/accelio existing, or we need to make sure the 'make dist' tarball includes the (possibly empty) directory.

**Comment by liewegas** [comment]
wip-xio is failing the gitbuilders (i think) because the 'make dist' tarball is done without --with-xio and so the src/accelio is not part of the tarball.  then the build fails with

src/Makefile.am:42: `src/test/Makefile.am' included from here
src/Makefile.am:4: required directory src/accelio does not exist

**Comment by liewegas** [comment]
So, the annoying thing is we can't just ./configure --with-xio and then make dist because the host may not have the accelio dependencies (ibverbs and whatnot).  I think that means we need to do something like Makefile-rocksdb.am, which is pretty annoying to maintain.  See https://github.com/ceph/ceph/blob/master/src/Makefile-rocksdb.am

**Comment by liewegas** [comment]
@mattbenjamin ping

**Comment by liewegas** [comment]
Any better ideas on how to deal with this in general?  :(

**Comment by dachary** [comment]
@liewegas the better solution is to package / backport the library so that it does not need to be included as an optional submodule. But that takes time and in the meantime we have to live with the imperfect solution.

**Comment by mattbenjamin** [comment]
What behaviors are actually required?

**Comment by liewegas** [comment]
'make dist' on an default config (w/ minimal dependencies present) should generate a tarball that has all source to build with the additional optional stuff...

**Comment by liewegas** [comment]
@mattbenjamin ping

**Comment by liewegas** [comment]
closing this for now, @mattbenjamin ... feel free to reopen

**Pull request closed without merging**